backend/src/endpoints/user/google.ts

```diff
@@ -0,0 +1,77 @@
+import { Request, Response } from "express";
+import {
+    handleGoogleLoginRequest,
+    verifyGoogleToken,
+    updateAlgoToken,
+    verifyAlgoToken,
+} from "../../services/authservice";
+
+export const googleLogin = async (req: Request, res: Response) => {
+    // get google token, verify it, create AlgoToken for user
+    const token = req.params.token;
+    const nickname = req.body.nickname;
+
+    const payload = await verifyGoogleToken(token);
+
+    if (!payload) {
+        res.status(400).send("Invalid Google token");
+        return;
+    }
+
+    const AlgoToken = await handleGoogleLoginRequest(payload, nickname);
+
+    if (!AlgoToken) {
+        res.status(500).send();
+        return;
+    }
+
+    res.cookie("AlgoToken", AlgoToken, { httpOnly: true }); // secure: true for https
```

Comment:
Tutaj złożyłbym prośbę do CEO o HTTPS na mikr.us, bo w 2022 nie wypada wysyłać tokenów (niczego, ale tokenów w szczególnosci!) po HTTP. (a samą tą linię zaznaczył jako TODO)

Reply:
Rozważę prośbę xD A tak serio no to jak znajdę czas to postaram się ogarnąć https jakąś domenę itp. Ewentualnie jak będzie ktoś chętny to dam mu uprawnienia i będzie mógł to ogarnąć. Co do samego tokenu to on i tak jest chyba szyfrowany po stronie backendu, i żeby go odszyfrować trzeba znać ALGO_SECRET.

Reply:
+ Rozumiem, że tylko w 2022 nie wypada wysyłać tokenów po HTTP, a w 2023 już wypada? xD